src/pgduckdb_background_worker.cpp

```diff
@@ -329,18 +333,59 @@ CreateTable(const char *postgres_schema_name, const char *table_name, const char
 		 * It's an old version of this DuckDB table, we can safely
 		 * drop it and recreate it.
 		 */
-		auto drop_query = DropPgTableString(postgres_schema_name, table_name, drop_with_cascade);
-		create_table_query = psprintf("%s %s", drop_query.c_str(), create_table_query);
+		std::string drop_table_query = DropPgTableString(postgres_schema_name, table_name, drop_with_cascade);
+
+		/* We use this to roll back the drop if the CREATE after fails */
+		BeginInternalSubTransaction(NULL);
+
+		/* Revert back to original privileges */
+		if (!SPI_run_utility_command(drop_table_query.c_str())) {
+
+			ereport(WARNING, (errmsg("Failed to sync MotherDuck table %s.%s", postgres_schema_name, table_name),
+
+			                  errdetail("While executing command: %s", create_table_query),
+			                  errhint("See previous WARNING for details")));
+			/*
+			 * Rollback the subtransaction to clean up the subtransaction
+			 * state. Even though there's nothing actually in it. So we could
+			 * we could just as well commit it, but rolling back seems more
+			 * sensible.
+			 */
+			RollbackAndReleaseCurrentSubTransaction();
+			return false;
+		}
+
+		did_delete_table = true;
 	}
 
-	if (!SPI_run_utility_command(create_table_query)) {
+	Oid saved_userid;
+	int sec_context;
+	GetUserIdAndSecContext(&saved_userid, &sec_context);
+	SetUserIdAndSecContext(MotherDuckPostgresUser(), sec_context | SECURITY_LOCAL_USERID_CHANGE);
```

Comment:
I assume there's no problem if `saved_userid==MotherDuckPostgresUser()` and there's no problem doing it because it's a trivial call?